src/core/time.d

```diff
@@ -1667,6 +1667,52 @@ public:
         }
     }
 
+    /++
+        Converts `Duration` to a `timeval` structure.
+    +/
+    timeval toTimeVal() const nothrow @nogc
+    {
+        timeval value;
+        split!("seconds", "usecs")(value.tv_sec, value.tv_usec);
+        return value;
+    }
+
+    unittest
+    {
+        timeval value;
+        value.tv_sec = 1;
+        value.tv_usec = 5000;
+        assert(dur!"msecs"(1005).toTimeVal() == value);
+    }
+
+    version(StdDdoc)
```

Comment:
In druntime, you need to use `CoreDdoc`. (Look at the rest of the file for examples.)